File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    const wordRef = [];
    const ref = "AEIOU";
    
    const dfs = (result) => {
        if (result.length) wordRef.push(result);
        
        if (result.length === 5) return;
        
        for (let i = 0; i < ref.length; i++) {
            dfs(result + ref[i]);
        }
    }
    
    dfs("");
    
    return wordRef.indexOf(word) + 1;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    alphabet = ["A", "E", "I", "O", "U"]
    dictionary = []
    flag = False
    
    def dfs(check_word, word, dictionary):
        nonlocal flag
        if check_word not in dictionary:
            dictionary.append(check_word)
        
        if check_word == word:
            flag = True
            return
        
        if len(check_word) == 5:
            return
        
        for item in alphabet:
            if not flag:
                dfs(check_word + item, word, dictionary)
    
    dfs("", word, dictionary)

    return dictionary.index(word)